c error: Mutable receiver reassignment with spread operator

**OS:** linux, Ubuntu 20.04 LTS
**Processor:** 8 cpus, 64bit, little endian, Intel(R) Core(TM) i5-8250U CPU @ 1.60GHz
**CC version:** cc (Ubuntu 9.3.0-17ubuntu1~20.04) 9.3.0

**vexe mtime:** 2020-11-30 06:51:51
**is vroot writable:** true
**V full version:** V 0.1.30 aae5812

**Git version:** git version 2.25.1
**.git/config present:** true
**thirdparty/tcc status:** thirdparty-linux-amd64 7543de81

**What did you do?**
```
module main

struct Person {
	name string
}

fn (mut p Person) change(name string) {
	p = { p | name: name }
}

fn main() {
	mut p := Person{'bob'}
        p.change('tom')
}
```

**What did you expect to see?**
Compiler error (??)
 
**What did you see instead?**
C compiler error:
```
recompilation with tcc failed; retrying with cc ...
==================
/tmp/v/test.9566610808835751915.tmp.c: In function ‘main__Person_change’:
/tmp/v/test.9566610808835751915.tmp.c:8066:3: error: field name not in record or union initializer
 8066 |   .name = name,
      |   ^
/tmp/v/test.9566610808835751915.tmp.c:8066:3: note: (near initialization for ‘(anonymous)’)
/tmp/v/test.9566610808835751915.tmp.c:8066:11: error: incompatible types when initializing type ‘main__Person *’ {aka ‘struct main__Person *’} using type ‘string’ {aka ‘struct string’}
 8066 |   .name = name,
      |           ^~~~
/tmp/v/test.9566610808835751915.tmp.c:8063:56: warning: parameter ‘p’ set but not used [-Wunused-but-set-parameter]
 8063 | VV_LOCAL_SYMBOL void main__Person_change(main__Person* p, string name) {
      |                                          ~~~~~~~~~~~~~~^
...
```

--- a/vlib/v/checker/struct.v
+++ b/vlib/v/checker/struct.v
@@ -548,7 +548,7 @@ pub fn (mut c Checker) struct_init(mut node ast.StructInit) ast.Type {
 	if node.has_update_expr {
 		update_type := c.expr(node.update_expr)
 		node.update_expr_type = update_type
-		if c.table.type_kind(update_type) != .struct_ {
+		if c.table.sym(update_type).kind != .struct_ {
 			s := c.table.type_to_str(update_type)
 			c.error('expected struct, found `$s`', node.update_expr.pos())
 		} else if update_type != node.typ {

--- a/vlib/v/tests/struct_init_update_with_mutable_receiver_test.v
+++ b/vlib/v/tests/struct_init_update_with_mutable_receiver_test.v
@@ -1,0 +1,21 @@
+module main
+
+struct Person {
+	name string
+	age  int
+}
+
+fn (mut p Person) change(name string) {
+	p = Person{
+		...p
+		name: name
+	}
+}
+
+fn test_struct_init_update_with_mutable_receiver() {
+	mut p := Person{'bob', 22}
+	p.change('tom')
+	println(p)
+	assert p.name == 'tom'
+	assert p.age == 22
+}

